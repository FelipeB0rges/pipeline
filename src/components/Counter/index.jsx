// Counter.js

import React, { useState } from 'react';

const Counter = () => {
  const [count, setCount] = useState(0);

  const incrementCount = () => {
    setCount(count + 1);
  };

  return (
    <div style={{
        color: "white",
    }}>
      <h1>Counter: {count}</h1>
      <button onClick={incrementCount}>Increment</button>
    </div>
  );
};

export default Counter;