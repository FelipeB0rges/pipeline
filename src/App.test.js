import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Counter from './Counter';

test('renders counter value', () => {
  render(<Counter />);
  const counterElement = screen.getByText(/counter: 0/i);
  expect(counterElement).toBeInTheDocument();
});

test('increments counter value on button click', () => {
  render(<Counter />);
  const counterElement = screen.getByText(/counter: 0/i);
  const buttonElement = screen.getByRole('button', { name: /increment/i });

  fireEvent.click(buttonElement);

  expect(counterElement).toHaveTextContent(/counter: 1/i);
});