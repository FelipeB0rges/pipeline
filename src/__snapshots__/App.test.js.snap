// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component renders InputComponent correctly 1`] = `
<div>
  <div>
    <h1>
      Test Input
    </h1>
    <input
      data-testid="input-component"
      type="text"
    />
  </div>
</div>
`;

exports[`Test Component renders ListComponent correctly 1`] = `
<div>
  <div>
    <h1>
      Test List
    </h1>
    <ul
      data-testid="list-component"
    >
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
      <li>
        Item 3
      </li>
    </ul>
  </div>
</div>
`;

exports[`Test Component renders ListComponent correctly 2`] = `
<div>
  <div>
    <h1>
      Test List
    </h1>
    <ul
      data-testid="list-component"
    >
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
      <li>
        Item 3
      </li>
    </ul>
  </div>
</div>
`;

exports[`Test Component should render Home component correctly 1`] = `
<div>
  <div
    class="versao"
    style="color: white;"
  >
    V: 1.0.0
  </div>
  <div
    class="home"
  >
    <div
      class="git"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSize40px css-821wyw-MuiSvgIcon-root"
        data-testid="GitHubIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
        />
      </svg>
      <div
        class="nome"
      >
        FelipeB0rges
      </div>
    </div>
  </div>
  <div
    class="test"
  >
    <div
      style="color: white;"
    >
      <h1>
        Counter: 
        0
      </h1>
      <button>
        Increment
      </button>
    </div>
    <div>
      <h1>
        Test Button
      </h1>
      <button
        data-testid="button-component"
      >
        Click me
      </button>
    </div>
    <div>
      <h1>
        Test Select
      </h1>
      <select
        data-testid="select-component"
      >
        <option
          value="option1"
        >
          Option 1
        </option>
        <option
          value="option2"
        >
          Option 2
        </option>
        <option
          value="option3"
        >
          Option 3
        </option>
      </select>
    </div>
    <div>
      <h1>
        Test Checkbox
      </h1>
      <input
        data-testid="checkbox-component"
        type="checkbox"
      />
    </div>
    <div>
      <h1>
        Test List
      </h1>
      <ul
        data-testid="list-component"
      >
        <li>
          Item 1
        </li>
        <li>
          Item 2
        </li>
        <li>
          Item 3
        </li>
      </ul>
    </div>
    <div>
      <h1>
        Test Input
      </h1>
      <input
        data-testid="input-component"
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`Testar Componente deve renderizar o componente Home corretamente 1`] = `
<div>
  <div
    class="versao"
    style="color: white;"
  >
    V: 1.0.0
  </div>
  <div
    class="home"
  >
    <div
      class="git"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSize40px css-821wyw-MuiSvgIcon-root"
        data-testid="GitHubIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
        />
      </svg>
      <div
        class="nome"
      >
        FelipeB0rges
      </div>
    </div>
  </div>
  <div
    class="test"
  >
    <div
      style="color: white;"
    >
      <h1>
        Counter: 
        0
      </h1>
      <button>
        Increment
      </button>
    </div>
    <div>
      <h1>
        Test Button
      </h1>
      <button
        data-testid="button-component"
      >
        Click me
      </button>
    </div>
    <div>
      <h1>
        Test Select
      </h1>
      <select
        data-testid="select-component"
      >
        <option
          value="option1"
        >
          Option 1
        </option>
        <option
          value="option2"
        >
          Option 2
        </option>
        <option
          value="option3"
        >
          Option 3
        </option>
      </select>
    </div>
    <div>
      <h1>
        Test Checkbox
      </h1>
      <input
        data-testid="checkbox-component"
        type="checkbox"
      />
    </div>
    <div>
      <h1>
        Test List
      </h1>
      <ul
        data-testid="list-component"
      >
        <li>
          Item 1
        </li>
        <li>
          Item 2
        </li>
        <li>
          Item 3
        </li>
      </ul>
    </div>
    <div>
      <h1>
        Test Input
      </h1>
      <input
        data-testid="input-component"
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`Testar Componente renderiza InputComponent corretamente 1`] = `
<div>
  <div>
    <h1>
      Test Input
    </h1>
    <input
      data-testid="input-component"
      type="text"
    />
  </div>
</div>
`;

exports[`Testar Componente renderiza ListComponent corretamente 1`] = `
<div>
  <div>
    <h1>
      Test List
    </h1>
    <ul
      data-testid="list-component"
    >
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
      <li>
        Item 3
      </li>
    </ul>
  </div>
</div>
`;

exports[`Testar Componente renderiza ListComponent corretamente 2`] = `
<div>
  <div>
    <h1>
      Test List
    </h1>
    <ul
      data-testid="list-component"
    >
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
      <li>
        Item 3
      </li>
    </ul>
  </div>
</div>
`;
